### PR TITLE
[RW-8327][risk=no] Handle the runtime error state on notebook pages

### PR DESCRIPTION
**Pull request by calbach**
Due to a regression ~8 months ago, when a runtime hit the error state the preview and notebook redirect pages would simply continue as if the runtime were ready to go. This results in a bunch of spurious 500 errors, and a confusing UX. Note that when a user does hit this scenario, they will also get a pop-up modal. This screen is mostly background, and is what they will see upon closing the modal.

https://user-images.githubusercontent.com/822298/170149067-dcc6aff5-219d-497f-b069-19e2ca3309f1.mp4

### CODE REVIEW

File: ui/src/app/pages/analysis/interactive-notebook.tsx [hunk 1]
```diff
@@ -200,8 +207,8 @@ export const InteractiveNotebook = fp.flow(
         );
         onRuntimeReady();
       } catch (e) {
-        this.setState({ userRequestedExecutableNotebook: false });
         if (e instanceof InitialRuntimeNotFoundError) {
```
Comment (evrii) [inline ui/src/app/pages/analysis/interactive-notebook.tsx:210]: Are there any other errors that could prevent the user from proceeding?

Comment (calbach) [reply ui/src/app/pages/analysis/interactive-notebook.tsx]: Thanks for pointing this out, I actually didn't mean to leave this change in. I wound up going with an approach that didn't depend on this boolean. Instead, the error has priority in rendering the top-bar status.

Comment (calbach) [reply ui/src/app/pages/analysis/interactive-notebook.tsx]: Let me know tomorrow if this resolved your question here - I'm not in a rush to merge this

Comment (evrii) [reply ui/src/app/pages/analysis/interactive-notebook.tsx]: This seems good to me.

**Pull request merged**